test/pbt/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+set(COMPILER_ARGS
+    --std=c++${CMAKE_CXX_STANDARD}
+    -Wall
+    -Wcast-align
+    -Wconversion
+    -Wdouble-promotion
+    -Werror
+    -Wextra
+    -Wextra-semi
+    -Wfatal-errors
+    -Wformat=2
+    -Wold-style-cast
+    -Woverloaded-virtual
+    -Wshadow
+    -Wunused
+    -Wno-missing-braces)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    list(
+        APPEND
+        COMPILER_ARGS
+        -fbracket-depth=99999
+        -Warray-bounds-pointer-arithmetic
+        -Wextra-semi-stmt
+        -Wgcc-compat
+        -Wheader-hygiene
+        -Widiomatic-parentheses
+        -Wimplicit
+        -Wnewline-eof
+        -Wmissing-prototypes
+        -Wno-gnu-string-literal-operator-template)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    list(APPEND COMPILER_ARGS -Wduplicated-branches -Wduplicated-cond
+         -Wlogical-op -Wuseless-cast)
+endif()
```

Comment:
**suggestion**: We can pull out the compiler-specific warnings from the `warnings` target with something like:
```cmake
get_target_property(specific_warnings warnings INTERFACE_COMPILE_OPTIONS) 
list(FILTER specific_warnings INCLUDE REGEX ${CMAKE_CXX_COMPILER_ID})
list(TRANSFORM specific_warnings REPLACE ".*:(-.*)>$" "\\1" OUTPUT_VARIABLE specific_warnings)
list(REMOVE_ITEM specific_warnings "")
```
This isn't perfect at the moment because we have one generator expression which depends on the version of clang, but we can change that up. Or alternatively we could evaluate the warnings (including genexes) to a file and have the python read it?

Reply:
How could we evaluate the warnings to a file? That would work well.

Reply:
We can use `file(GENERATE ...)` to write a file, for example:
```cmake
get_target_property(warn_list warnings INTERFACE_COMPILE_OPTIONS)
string(REPLACE ";" "," warn_list "${warn_list}")
file(
    GENERATE
    OUTPUT "${CMAKE_BINARY_DIR}/warnings.txt"
    CONTENT "${warn_list}" TARGET warnings)
```
This will produce a file with contents like:
```
-Wall,-Warray-bounds-pointer-arithmetic,-Wcast-align,-Wconversion,-Wdouble-promotion,,,-Werror,-Wextra,-Wextra-semi,-Wextra-semi-stmt,-Wfatal-errors,-Wformat=2,-Wgcc-compat,-Wheader-hygiene,-Widiomatic-parentheses,-Wimplicit,,-Wnewline-eof,-Wold-style-cast,-Woverloaded-virtual,-Wshadow,,-Wunused,-Wmissing-prototypes,-Wno-gnu-string-literal-operator-template
```
There are some empty strings in there but that should not be a problem.